modules/mesh-layers/src/scenegraph-layer/scenegraph-layer-uniforms.ts

```diff
@@ -0,0 +1,35 @@
+import type {Texture} from '@luma.gl/core';
+import type {Matrix4} from '@math.gl/core';
+import {ShaderModule} from '@luma.gl/shadertools';
+import {UniformTypes} from '@deck.gl/core';
+
+const uniformBlock = `\
+uniform scenegraphUniforms {
+  float sizeScale;
+  float sizeMinPixels;
+  float sizeMaxPixels;
+  mat4 sceneModelMatrix;
+  bool composeModelMatrix;
+} scenegraph;
+`;
+
+export type ScenegraphProps = {
+  sizeScale: number;
+  sizeMinPixels: number;
+  sizeMaxPixels: number;
+  sceneModelMatrix: Matrix4;
+  composeModelMatrix: boolean;
+};
+
+export const scenegraphUniforms = {
+  name: 'scenegraph',
+  vs: uniformBlock,
+  fs: uniformBlock,
+  uniformTypes: {
+    sizeScale: 'f32',
+    sizeMinPixels: 'f32',
+    sizeMaxPixels: 'f32',
+    sceneModelMatrix: 'mat4x4<f32>',
+    composeModelMatrix: 'f32'
+  } as const satisfies UniformTypes<ScenegraphProps>
```

Comment:
Q: Can this not be made implicit from the outer `as const satsifies`? Or maybe for another PR.

Reply:
At a quick test – I think the outer cast is enough, it's catching errors within the `uniformTypes` block as well.

Reply:
Removing `UniformTypes<ScenegraphProps>` will not catch an error like `sizeMaxPixels: 'vec2<f32>'` as that is a valid `UniformFormat`, but it doesn't match the type of `ScenegraphProps["sizeMaxPixels"]`.

Arguably we could move the `UniformTypes` type to luma, however it is not a simple as this example suggests as the types `T` & `S` in `ShaderModule<T>` & `UniformTypes<S>` are not always the same, e.g. in [bitmapUniforms](https://github.com/visgl/deck.gl/blob/master/modules/layers/src/bitmap-layer/bitmap-layer-uniforms.ts#L39-L40).

I don't think it hurts to keep this here as it make it clear what is going on, rather than the reader having to understand how the generics work in `ShaderModule`